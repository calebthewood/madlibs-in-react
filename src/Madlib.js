import React from "react";


function Madlib({ title, generate, toggleForm }) {
  return (
    <div
      id="madlib"
      data-test-id="madlib"
      className="container">

      <h2>{title}</h2>

      <p id="content" className="lead">{generate()}</p>

      <button
        id="restart"
        className="btn btn-primary mb-3"
        onClick={toggleForm}
      >Restart</button>

    </div>
  );
}

export default Madlib;