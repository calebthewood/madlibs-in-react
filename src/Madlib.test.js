import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Madlib from "./Madlib";


describe("Madlib Tests", () => {
    /** mocks madlib content */
    function testContent(){
      return "testing madlib content.";
    }

  /** Smoke Test */
  it("renders without crashing", function () {
    render(
      <Madlib
        generate={jest.fn}
        toggleForm={jest.fn}
      />);
  });

  /** Snapshot Test */
  it("snapshot", function () {
    const { container } = render(
      <Madlib
        generate={jest.fn}
        toggleForm={jest.fn}
      />);
    expect(container).toMatchSnapshot();
  });

  test("renders content", () => {
    const { container } = render(
      <Madlib
        generate={testContent}
        toggleForm={jest.fn}
      />);

      expect.assertions(2);
      expect(container.querySelector("#madlib").dataset.testId)
        .toEqual("madlib");
      expect(container.querySelector("#content").innerHTML)
        .toEqual("testing madlib content.")
    });

    test("renders restart button", () => {
      const { container } = render(
        <Madlib
          generate={testContent}
          toggleForm={jest.fn}
        />);

        expect(container.querySelector("#restart")).toBeInTheDocument()
      });
});