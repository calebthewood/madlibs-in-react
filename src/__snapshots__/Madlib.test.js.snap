// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Madlib Tests snapshot 1`] = `
<div>
  <div
    class="container"
    data-test-id="madlib"
    id="madlib"
  >
    <h2 />
    <p
      class="lead"
      id="content"
    />
    <button
      class="btn btn-primary mb-3"
      id="restart"
    >
      Restart
    </button>
  </div>
</div>
`;
